tls/s2n_config.c

```diff
@@ -241,10 +241,12 @@ int s2n_config_defaults_init(void)
         POSIX_GUARD(s2n_config_load_system_certs(&s2n_default_config));
     }
 
-    /* Set up TLS 1.3 defaults */
-    POSIX_GUARD(s2n_config_init(&s2n_default_tls13_config));
-    POSIX_GUARD(s2n_config_setup_tls13(&s2n_default_tls13_config));
-    POSIX_GUARD(s2n_config_load_system_certs(&s2n_default_tls13_config));
+    /* TLS 1.3 default config is only used in tests so avoid initialization costs in applications */
+    if (s2n_in_test()) {
```

Comment:
Why not just move the if down around s2n_config_load_system_certs so that there's no way we end up without a config?

Reply:
I debated doing that but my worry with that is will potentially lead to a config in an unexpected state. For example, say someone goes to expose this config outside of tests in the future. I think I'd rather have it segfault in integrating that rather than the system certs silently not being there, as they might expect.

Reply:
we ended up removing the `s2n_in_test` branch entirely and just skip the certs for this config.